gems/pending/VMwareWebService/MiqVimInventory.rb

```diff
@@ -1899,10 +1899,11 @@ def inventoryHash_locked
       @cacheLock.sync_lock(:EX) if (unlock = @cacheLock.sync_shared?)
 
       $vim_log.info "MiqVimInventory(#{@server}, #{@username}).inventoryHash_locked: calling retrieveProperties" if $vim_log
-      rv = retrieveProperties(@propCol, @spec)
+      rra = retrievePropertiesEx(@propCol, @spec)
+      oca = rra['objects'] if rra
```

Comment:
These 2 lines seem to be a reoccurring pattern. Maybe you should add a convenience method to VIM for clients that expect the old `retrieveProperties` behavior. Something like `retrievePropertiesCompat` might be good. It would call `retrievePropertiesEx` and return `rra['objects']`. Then you'd only have to change the method name in the clients.